### PR TITLE
Completed array-filter exercise. | Vera Yeh

### PR DESCRIPTION
**Pull request by VeraYeh**
image

### CODE REVIEW

File: array-filter/main.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  'Donna'
+];
+
+const evenNumbers = numbers.filter(number => number % 2 === 0);
+
+const overFive = numbers.filter(number => number > 5);
+
+const startWithE = names.filter(name => name[0] === 'E');
+
+const haveD = names.filter(name => name.toLowerCase().indexOf('d') !== -1);
```
Comment (scbowler) [inline array-filter/main.js:21]: Good job using `toLowerCase` 👍

**Review by scbowler — APPROVED** [review]
Looks good! 👍 

Take a shot at the "Bonus Challenge" now

**Comment by VeraYeh** [comment]
Added bonus challenge
image

File: array-filter/filter.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  'Joan',
+  'Elizabeth',
+  'Janese',
+  'Donna'
+];
+
+function filters(array, predicate) {
+  const newArray = [];
+  for (let i = 0; i < array.length; i++) {
+    if (predicate(array[i]) === true) {
```
Comment (scbowler) [inline array-filter/filter.js:18]: This is fine but to make it a bit more flexible you can just check if the returned value is truthy by doing:

```JAVASCRIPT
if (predicate(array[i])) {
```

**Review by scbowler — APPROVED** [review]
BONUS CHALLENGE - Looks good 👍 

1 FYI comment, no changes required

**Pull request merged**